### DISCUSSION TITLE
Generating a Pixel / Bitmap Only Font

### DISCUSSION THREAD

**Discussion by dcrawl** [fontforge/fontforge#4842] [Q&A]
Hi,

I'm trying to create a Bitmap Only font for a pixel art game.  I followed the information about creating Bitmap Strikes to create my bitmap font.  I don't have a bitmap for everything but I don't think that matters from what I read.  When I try to Generate the font, my font file is empty.  It doesn't display anything.  I'm exporting as No Outline and Fake MS bitmap only.  I have tried other combinations and nothing seems to come out.  I'm not certain what I did wrong and I can't find a tutorial specially for bitmap font generation.

Any help would be appreciated.  Thanks.

James

**Comment by skef** [comment]
Is the generated font file "empty" when you try opening it in FontForge itself? 

What type of bitmap font are you trying to generate? Some platforms have dropped support for most bitmap font formats.

**Reply by dcrawl** [reply]
The Bitmap strikes are there when I open it in FontForge but the font previewer has nothing in it and Font Book on my Mac shows an empty preview.

I'm not sure which format to export as so that it will work.

**Comment by ctrlcctrlv** [comment]
Such fonts are essentially useless in modern practice. I recommend use of my [bitmapfont2otb](https://github.com/ctrlcctrlv/bitmapfont2otb) script; it takes input of a filename referring to some bitmap font FontForge understands (even if in SFD) and gives you an "OpenType Bitmap" font which has the pixels as squares in the CFF table and all the bitmap strikes as well. You should configure your applications to correctly rasterize bitmap fonts if it's not being rasterized correctly.

**Reply by dcrawl** [reply]
It may be useless in modern practises but it is useful in Game Development and you can find many bitmap fonts being sold for it.  At low resolutions, vector fonts do not render well and bitmap fonts are preferred.  I'll take a look at the script.  It sounds like it may help.  Thanks.

J

**Reply by ctrlcctrlv** [reply]
OK, I just released v2. I realize what you are saying is true, thus the script. I'm making a point about _format_ rather than about _æsthetics_. Bitmap-only formats are nearly useless, bitmap alongside vectors in OpenType is the best solution, and Pango supports only that as of last year.
